src/Mod/Draft/importDXF.py

```diff
@@ -2724,6 +2739,10 @@ def processdxf(document, filename, getShapes=False, reComputeFlag=True):
                         formatObject(newob, insert)
             num += 1
 
+    # Move layer contents to layers
+    for (l, contents) in layerObjects.items():
+        l.Group = contents
```

Comment:
This code is problematic if the layer already exists and contains objects (from a previous import for example).

Reply:
Ooh, good point. Same applies to the C++ importer #16596. Will fix later today

Reply:
Fixed. The assignment `l.Group = contents` was changed to `l.Group += contents`.